xbmcaddon.py

```diff
@@ -225,9 +252,48 @@ def __load_add_on_xml(self):
 
         with io.open(add_on_xml, encoding='utf-8') as fp:
             xml_content = fp.read()
-            self.__version = re.findall(r'version="(\d+.\d+.\d+[^"]*)', xml_content)[0]
+            self.__version = re.findall(r'<addon.*?version="(\d+(?:\.\d)?(?:\.\d)?[^"]*)', xml_content)[0]
             self.__add_on_id = re.findall(r'addon\W+id="([^"]+)"', xml_content)[0]
             self.__name = re.findall(r'name="([^"]+)"', xml_content)[0]
+            tmp = re.findall(r'<addon.*?provider-name="([^"]+)', xml_content)
```

Comment:
I never really like reusing a variable. So perhaps rename `tmp` to `author_matches` here? Also a `re.DOTALL` should be there.

Reply:
Ok, i will change every tmp variable.